lib/src/utils.dart

```diff
@@ -6,8 +6,8 @@ import 'package:source_span/source_span.dart';
 
 /// A pair of values.
 class Pair<E, F> {
-  final E first;
-  final F last;
+  final E? first;
+  final F? last;
```

Comment:
It's hard to tell whether this is really necessary. It would be good to understand whether that's intentional, in which case the class comment should be updated to explain why, or accidental, in which case the fields should be made non-nullable and other code fixes as appropriate.

Reply:
+1, `E` and `F` are not bound to `Object` anyways so they already allow `null`

Reply:
Ok, I see where I goofed here.  Fixing up. 👍